SimpleServer/Source/ServerImitator.cpp

```diff
@@ -6,140 +6,147 @@
-			continue;
-		}
-
-		if (const auto [value, check] = utils::parseCoordinateSystem(dataBuffer); check)
-		{
-			const std::string coordSystemStr = dataBuffer.substr(0u, 1u);
-			_printer.writeLine(std::cout, coordSystemStr);
-			_coorninateSystem.emplace(value);
-		}
-
-		_logger.writeLine(_messageWithIP, '-', dataBuffer);
-		std::string toSending = utils::parseFullData(dataBuffer);
-
-		bool flag;
-		const auto robotData = utils::fromString<RobotData>(dataBuffer, flag);
-		if (flag)
-		{
-			std::this_thread::sleep_for(calculateDuration(robotData));
-		}
-
-		if (!toSending.empty())
-		{
-			sendData(_clientSendingSocket, toSending);
-		}
-	}
+    _printer.writeLine(std::cout, "\n\n\nWaiting for reply...\n");
+
+    waitingForConnections();
+
+    _logger.writeLine("Server started to receive at", utils::getCurrentSystemTime());
+
+    while (true)
+    {
+
+        const auto [dataBuffer, status] = receiveData(_clientReceivingSocket, _messageWithIP,
+                                                      _buffer);
+        _isRunning.store(status);
+
+        if (!_isRunning.load())
+        {
+            waitingForConnections();
+            continue;
+        }
+
+        if (const auto [value, check] = utils::parseCoordinateSystem(dataBuffer); check)
+        {
+            const std::string coordSystemStr = dataBuffer.substr(0u, 1u);
+            _printer.writeLine(std::cout, coordSystemStr);
+            _coorninateSystem.emplace(value);
+        }
+
+        _logger.writeLine(_messageWithIP, '-', dataBuffer);
+        std::string toSending = utils::parseFullData(dataBuffer);
+
+        bool flag;
+        const auto robotData = utils::fromString<RobotData>(dataBuffer, flag);
+        if (flag)
+        {
+            std::this_thread::sleep_for(calculateDuration(robotData));
+        }
+
+        if (!toSending.empty())
+        {
+            sendData(_clientSendingSocket, toSending);
+        }
+    }
 }
 
 void ServerImitator::run()
 {
-	_isRunning.store(true);
-	waitLoop();
+    _isRunning.store(true);
+    waitLoop();
 }
 
 void ServerImitator::launch()
 {
-	bindSocket(_sendingSocket, _sendingSocketAddress, _clientSendingPort);
-	bindSocket(_receivingSocket, _receivingSocketAddress, _clientReceivingPort);
+    bindSocket(_sendingSocket, _sendingSocketAddress, _clientSendingPort);
+    bindSocket(_receivingSocket, _receivingSocketAddress, _clientReceivingPort);
 
-	listenOn(_sendingSocket, _backlog);
-	listenOn(_receivingSocket, _backlog);
+    listenOn(_sendingSocket, _backlog);
+    listenOn(_receivingSocket, _backlog);
 }
 
 void ServerImitator::waitingForConnections()
 {
-	closeSocket(_clientReceivingSocket);
-	closeSocket(_clientSendingSocket);
-	_isRunning.store(false);
-	process();
+    closeSocket(_clientReceivingSocket);
+    closeSocket(_clientSendingSocket);
+    _isRunning.store(false);
+    process();
 }
 
 std::chrono::milliseconds ServerImitator::calculateDuration(const RobotData& robotData)
```

Comment:
May be use delay manager?

Reply:
No, imitator works without any statistics, only uses static formula. `DelayManager` is needed in file with data.